ipatests/test_integration/test_membermanager.py

```diff
@@ -212,3 +212,73 @@ def test_hostgroup_member_manager_nopermission(self):
             "'write' privilege to the 'memberManager' attribute of entry"
         )
         assert expected in result.stdout_text
+
+    def test_member_manager_upgrade_scenario(self):
+        """
+        Testing if manager whose rights defined by the group membership
+        is able to add group members, after upgrade of ipa server.
+        Using ACI modification to demonstrate unability before upgrading
+        ipa server.
+
+        Related: https://pagure.io/freeipa/issue/9286
+        """
+        user = "idmuser"
+        password = "Secret123"
+        group1 = "role-groupmanager"
+        group2 = "role-usergroup-A"
+
+        master = self.master
+
+        tasks.kinit_admin(master)
+        tasks.group_add(master, group1)
+        tasks.group_add(master, group2)
+        tasks.create_active_user(master, user, password)
+
+        tasks.kinit_admin(master)
+        tasks.group_add_member(master, group1, user)
+        master.run_command(["ipa", "group-add-member-manager", "--groups",
+                            group1, group2])
+
+        base_dn = self.master.domain.basedn
+        aci_hostgroup = (
+            '(targetattr = "member")(targetfilter = '
+            '"(objectclass=ipaHostGroup)")'
+            '(version 3.0; acl "Allow member managers '
+            'to modify members of host groups"; allow (write) userattr = '
+            '"memberManager#USERDN" or userattr = "memberManager#GROUPDN";)'
+        )
+        aci_usergroup = (
+            '(targetattr = "member")(targetfilter = '
+            '"(objectclass=ipaUserGroup)")'
+            '(version 3.0; acl "Allow member managers '
+            'to modify members of user groups"; allow (write) userattr = '
+            '"memberManager#USERDN" or userattr = "memberManager#GROUPDN";)'
+        )
+        ldif_entry = tasks.textwrap.dedent(
+            """dn: cn=hostgroups,cn=accounts,{base_dn}
```

Comment:
Replace with 
```
      ldif_entry = tasks.textwrap.dedent(
            """
            dn: cn=hostgroups,cn=accounts,{base_dn}
            changetype: modify
            delete: aci
            aci: {aci_hostgroup}
            dn: cn=groups,cn=accounts,{base_dn}
            changetype: modify
            delete: aci
            aci: {aci_usergroup}
""").format(base_dn=base_dn,
                     aci_hostgroup=aci_hostgroup,
                     aci_usergroup=aci_usergroup)
```
As dedent removes the *common* leading spaces, the first line needs to have the same indentation as the remaining ones.

Reply:
This looks like it didn't work, or maybe my misunderstanding somewhere?

Reply:
My bad, I removed the empty line before the 2nd entry, should read instead:
```
      ldif_entry = tasks.textwrap.dedent(
            """
            dn: cn=hostgroups,cn=accounts,{base_dn}
            changetype: modify
            delete: aci
            aci: {aci_hostgroup}

            dn: cn=groups,cn=accounts,{base_dn}
            changetype: modify
            delete: aci
            aci: {aci_usergroup}
""").format(base_dn=base_dn,
                     aci_hostgroup=aci_hostgroup,
                     aci_usergroup=aci_usergroup)